src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -534,8 +534,15 @@ public Submission findLatestSubmissionWithRatedResultWithCompletionDate(Particip
         if (participation.getSubmissions() == null || participation.getSubmissions().isEmpty()) {
             return null;
         }
+
+        Result result = null;
+        List<Result> results;
+        List<Result> resultsOfLatestSubmission;
         for (var submission : participation.getSubmissions()) {
-            var result = submission.getResult();
+           results = submission.getResults();
```

Comment:
please create a method "getLatestResult()" in Submission and use this one instead of duplicating the code everywhere.
please also create a method "getLatestSubmissionResult()" in Participation that gets the last submission and the last result. Add JavaDoc that this one only works, if participation->submissions->results was loaded eagerly from the database

Reply:
further adaptions of those methods would be "getLatestResult(AssessmentType assessmentType)" and getLatestResult(boolean rated)"